packages/controller/src/lib/notificationHandler.ts

```diff
@@ -361,13 +456,16 @@ class NotificationHandler {
     /**
      * Returns the stored notifications matching the filters with description and name
      *
-     * @param {string|null|undefined} scopeFilter - scope of notifications
-     * @param {string|null|undefined} categoryFilter - category of notifications
-     * @param {string|null|undefined} instanceFilter - instance of notifications
-     * @return {object}
+     * @param scopeFilter - scope of notifications
+     * @param categoryFilter - category of notifications
+     * @param instanceFilter - instance of notifications
      */
-    getFilteredInformation(scopeFilter, categoryFilter, instanceFilter) {
-        const res = {};
+    getFilteredInformation(
+        scopeFilter: string | null | undefined,
+        categoryFilter: string | null | undefined,
+        instanceFilter: string | null | undefined
+    ): FilteredNotificationInformation {
+        const res: Partial<FilteredNotificationInformation> = {};
```

Comment:
```suggestion
        const res: FilteredNotificationInformation = {};
```
The `Partial` makes your life below harder because it adds `| undefined` to the type of the properties.